Source/CombatExtended/CombatExtended/StatParts/StatPart_QualityConditional.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Verse;
+using RimWorld;
+using UnityEngine;
+
+namespace CombatExtended
+{
+	public class StatPart_QualityConditional : StatPart
+	{
+		private bool applyToNegativeValues = false;
+
+		private float factorAwful = 1f;
+
+		private float factorPoor = 1f;
+
+		private float factorNormal = 1f;
+
+		private float factorGood = 1f;
+
+		private float factorExcellent = 1f;
+
+		private float factorMasterwork = 1f;
+
+		private float factorLegendary = 1f;
+
+		private float maxGainAwful = 9999999f;
+
+		private float maxGainPoor = 9999999f;
+
+		private float maxGainNormal = 9999999f;
+
+		private float maxGainGood = 9999999f;
+
+		private float maxGainExcellent = 9999999f;
+
+		private float maxGainMasterwork = 9999999f;
+
+		private float maxGainLegendary = 9999999f;
+
+		public override void TransformValue(StatRequest req, ref float val)
+		{
+			if (req.Def.GetModExtension<QualityConditionalModExt>()?.UseQualityMult ?? true)
+			{
+				if (!(val <= 0f) || applyToNegativeValues)
+				{
+					float a = val * QualityMultiplier(req.QualityCategory) - val;
+					a = Mathf.Min(a, MaxGain(req.QualityCategory));
+					val += a;
+				}
+			}
+
+		}
+
+		public override string ExplanationPart(StatRequest req)

```

Comment:
Similarly in this method, you've already started w/ early returns, keep at it.

The opposite of `if (optional?.shit() ?? true)` is `if (!optional?.shit() ?? false)`